src/Silicon.cpp

```diff
@@ -226,10 +534,18 @@ namespace galsim {
 #pragma omp parallel for
 #endif
         for (size_t k=0; k<_imagepolys.size(); ++k) {
-            if (changed[k]) _imagepolys[k].updateBounds();
+            if (changed[k]) {
+                _imagepolys[k].updateBounds();
+                updatePixelBounds(nx, ny, k);
```

Comment:
I just checked and unfortunately the new version is slower here (maybe 1.5x slower on average). It should be possible to improve it. If necessary we can use a custom loop for this instead of `iteratePixelBoundary`, though it will be less readable.